### DISCUSSION TITLE
Understanding val.py --task study output

### DISCUSSION THREAD

**Discussion by pasmai** [ultralytics/yolov5#6529] [Q&A]
I used `val.py --task study weights w1.pt w2.pt w3.pt --data data.yaml --iou 0.7`

and it wrote the following output. Can anybody help me what exactly those numbers mean.
And how do I know what imgsize and which model is on which row? I feel like the script wrote a csv and forgot the header as well as the row names...

```csv
    0.9859     0.9686     0.9883     0.8803          0          0          0    0.03713      1.437     0.8805
    0.9976          1      0.995     0.9601          0          0          0    0.05902      1.385     0.7541
    0.9983          1      0.995     0.9706          0          0          0    0.07694      1.394     0.7212
    0.9960     0.9985      0.995     0.9663          0          0          0     0.1072      1.391     0.7185
    0.9972     0.9981      0.995     0.9342          0          0          0     0.1359      1.669     0.7431
    0.9723     0.9922     0.9821      0.858          0          0          0     0.1717       2.26     0.7472
    0.8981     0.9784     0.9803     0.7653          0          0          0     0.2238      2.901     0.8022
```
My guess so far:
`P, R, mAP@.5, mAP@.5:.95, t_preprocess (ms/img), t_inference (ms/img), t_NMS (ms/img)`

**Comment by glenn-jocher** [comment]
@pasmai ah yes good point, we should update the export format to CSV with column names. 

In the meantime you can plot your results with `utils.plots.plot_study()`:
https://github.com/ultralytics/yolov5/blob/19e0208fc9e33010717e066f9bd65c27db7c2b5c/utils/plots.py#L291-L326
